src/image-viewer.component.tsx

```diff
@@ -564,6 +571,8 @@ export default class ImageViewer extends React.Component<Props, State> {
             </ImageZoom>
           );
         case 'fail':
+          if (this.props.onError) this.props.onError(index);
```

Comment:
Perhaps [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) would work better?

Reply:
It would but I think optional chaining is more clear when you just try to extract a value and not when you want to call a method.

Looks more clear like it is now (however I would still keep the {} and not have the inline if statement)